posts/why-is-react-native-so-promising.md

```diff
@@ -0,0 +1,161 @@
+
+But  sometimes new languages yield new philosophies. And in this new era of webcomponents, HOCs are the next best thing. I'll try to explain the core principles while staying as concise as possible.
+
+#### HOC in a few words
+
+A Higher Order Component is essentially composed of one React component and one or many Higher Order Functions. A Higher Order Function is a function that accepts a function as a parameter and returns a React Component. In practice it will act as a component wrapper, you'll be able to add unrelated props or new mechanics; enriching the React component in any way you want. It is particularly interesting to hide logic, complexity or a concern that the Component shouldn't care about. 
+
+The most known Higher Order Function is the Redux `connect` function that allows you to bind the props of a "dumb" Component to the Redux Store. Here's a quick reminder of the function's profile:
+
+```
+connect(mapStateToProps, mapDispatchToProps)(Component)
+```
+
+I've deliberately left out the two other arguments of connect because they're not used as often as mapStateToProps and mapDispatchToProps.
+* `mapStateToProps` is a function that returns the mapping between the Component props and the Redux's store.
+* `mapDispatchToProps` is a function that returns the mapping between the Component props and the Actions.
+
+This is quite a special case because here connect is a function that will return the actual Higher Order Function. Let's break it down:
+
+```
+// Creating the Higher Order Function with mapStateToProps and mapDispatchToProps
+const hof = connect(mapStateToProps, mapDispatchToProps);
+// Creating the HOC with our Higher Order Function.
+const hoc = hof(MyComponent);
+```
+
+We now have a whole new component with the rendering capability of MyComponent while being linked to the Redux Component.
+
+On top of that, the HOC's philosophy is to use stateless components instead of the standard React Component class. A stateless component is a function that only returns JSX and takes some props as arguments. ES6 classes being only syntactic sugar for a language that doesn't have any object-oriented inheritance model, using them in React or React Native can lead to a heavier DOM generation (a div for the class that will wrap the render content).
+
+And, guess what, the best way to manage state in a stateless world is to use a Higher Order Component! Using the excellent [recompose](https://github.com/acdlite/recompose) library that provides a lot of React specific Higher Order Components, you can easily create reusable snippets for a lot of use cases. 
+
+Here's an example of a small Menu component: 
+<script src="https://gist.github.com/carvallegro/6167b34b0cdf9b6b19cb4358320d5856.js"></script>
```

Comment:
Change the script to a normal link?

Reply:
No this is to embed my gist into the page (avoid a huge bunk of code as it has scrolling and all)

Reply:
👍